src/Adapter/Product/QueryHandler/GetProductCombinationsForEditingHandler.php

```diff
@@ -0,0 +1,149 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationAttributeInformation;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationListForEditing;
+
+/**
+ * Handles @see GetProductCombinationsForEditing using legacy object model
+ */
+final class GetProductCombinationsForEditingHandler extends AbstractProductHandler implements GetProductCombinationsForEditingHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param ProductRepository $productRepository
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        ProductRepository $productRepository
+    ) {
+        $this->combinationRepository = $combinationRepository;
+        $this->productRepository = $productRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetProductCombinationsForEditing $query): CombinationListForEditing
+    {
+        $productId = $query->getProductId();
+        $this->productRepository->assertProductExists($productId);
+        $combinations = $this->combinationRepository->getProductCombinations($productId, $query->getLimit(), $query->getOffset());
+
+        $combinationIds = array_map(function ($combination): int {
+            return (int) $combination['id_product_attribute'];
+        }, $combinations);
+
+        $attributesInformation = $this->combinationRepository->getAttributesInfoByCombinationIds(
+            $combinationIds,
+            $query->getLanguageId()
+        );
+
+        return $this->formatCombinationsForEditing(
+            $combinations,
+            $attributesInformation,
+            $this->combinationRepository->getTotalCombinationsCount($productId)
+        );
+    }
+
+    /**
+     * @param array $combinations
+     * @param array<int, array<int, mixed>> $attributesInformationByCombinationId
+     * @param int $totalCombinationsCount
+     *
+     * @return CombinationListForEditing
+     */
+    private function formatCombinationsForEditing(
+        array $combinations,
+        array $attributesInformationByCombinationId,
+        int $totalCombinationsCount
+    ): CombinationListForEditing {
+        $combinationsForEditing = [];
+
+        foreach ($combinations as $combination) {
+            $combinationId = (int) $combination['id_product_attribute'];
+            $combinationAttributesInformation = [];
+
+            foreach ($attributesInformationByCombinationId[$combinationId] as $attributesInfo) {
+                foreach ($attributesInfo as $attributeInfo) {
+                    $combinationAttributesInformation[] = new CombinationAttributeInformation(
```

Comment:
In response to previous comment: To optimize the object creation (memory usage mainly) we could have only one instance of `CombinationAttributeInformation` per attribute_id and they would assigned/shared to the various `CombinationForEditing` that have those attributes in common

This would make sense when we have dozens of combinations to handle as it prevents duplicating this object for each combination that have it in common

But it's not mandatory for now